pkg/render/fluentd.go

```diff
@@ -1040,10 +1042,40 @@ func (c *fluentdComponent) eksLogForwarderDeployment() *appsv1.Deployment {
 		{Name: "AWS_REGION", Value: c.cfg.EKSConfig.AwsRegion},
 		{Name: "AWS_ACCESS_KEY_ID", ValueFrom: secret.GetEnvVarSource(EksLogForwarderSecret, EksLogForwarderAwsId, false)},
 		{Name: "AWS_SECRET_ACCESS_KEY", ValueFrom: secret.GetEnvVarSource(EksLogForwarderSecret, EksLogForwarderAwsKey, false)},
+		relasticsearch.ElasticIndexSuffixEnvVar(c.cfg.ESClusterConfig.ClusterName()),
```

Comment:
I think all of these are only needed in the env when EKS log forwarding is enabled, so we should only add them conditionally on that feature being enabled.

Reply:
These are added unconditionally but we're already inside the `eksLogForwarderDeployment` function which does the checks necessary to ensure we're only doing this in the case of EKS log forwarding being enabled.